src/timesync/timesyncd-bus.c

```diff
@@ -43,6 +45,60 @@ static int property_get_servers(
         return sd_bus_message_close_container(reply);
 }
 
+static int method_set_runtime_servers(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_strv_free_ char **valid_names = NULL;
+        _cleanup_strv_free_ char **msg_names = NULL;
+        Manager *m = userdata;
+        int r;
+
+        assert(m);
+        assert(message);
+
+        r = sd_bus_message_read_strv(message, &msg_names);
+        if (r < 0)
+                return r;
+
+        while (m->runtime_servers)
+                server_name_free(m->runtime_servers);
+
+        STRV_FOREACH(name, msg_names) {
+                r = dns_name_is_valid_or_address(*name);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to check validity of NTP server name or address '%s': %m", *name);
+
+                if (r == 0) {
+                        log_error("Invalid NTP server name or address, ignoring: %s", *name);
+                        continue;
+                }
+
+                strv_push(&valid_names, strdup(*name));
+        }
+
+        STRV_FOREACH(name, valid_names) {
+                r = server_name_new(m, NULL, SERVER_RUNTIME, *name);
+                if (r < 0) {
+                        while (m->runtime_servers)
+                                server_name_free(m->runtime_servers);
+
+                        return log_error_errno(r, "Failed to add runtime server '%s': %m", *name);
+                }
+        }
+
+        if (m->runtime_servers)
+                m->runtime_servers_changed = true;
```

Comment:
if an empty list was supplied this should also constitute a change, no? if something was set before…?

Reply:
Yes, that makes sense.